The job **`x86_64-gnu-gcc`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/26851786275/job/79185499700) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/79185499700) [(plain)](https://github.com/rust-lang/rust/commit/22e7e40b36b882a132b59776f3ba4607decd7adb/checks/79185499700/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
---- [ui] tests/ui/proc-macro/test.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/proc-macro/test.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "-Zcodegen-backend=gcc" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/proc-macro/test" "-A" "unused" "-W" "unused_attributes" "-A" "internal_features" "-A" "incomplete_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/proc-macro/test/auxiliary" "--edition=2021" "--extern" "proc_macro_api_tests=/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/proc-macro/test/auxiliary/libproc_macro_api_tests.so"
stdout: none
--- stderr -------------------------------
error: proc macro panicked
##[error]  --> /checkout/tests/ui/proc-macro/test.rs:11:1
   |
```

</details>

For more information how to resolve CI failures of this job, visit this [link](https://rustc-dev-guide.rust-lang.org/tests/codegen-backend-tests/cg_gcc.html).

--- a/library/proc_macro/src/lib.rs
+++ b/library/proc_macro/src/lib.rs
@@ -1648,7 +1648,7 @@ impl Literal {
     #[unstable(feature = "proc_macro_value", issue = "136652")]
     pub fn byte_character_value(&self) -> Result<u8, ConversionErrorKind> {
         self.0.symbol.with(|symbol| match self.0.kind {
-            bridge::LitKind::Char => unescape_byte(symbol)
+            bridge::LitKind::Byte => unescape_byte(symbol)
                 .map_err(|err| ConversionErrorKind::FailedToUnescape(err.into())),
             _ => Err(ConversionErrorKind::InvalidLiteralKind),
         })

--- a/tests/ui/proc-macro/auxiliary/api/literal.rs
+++ b/tests/ui/proc-macro/auxiliary/api/literal.rs
@@ -1,6 +1,6 @@
 // ignore-tidy-linelength
 
-use proc_macro::Literal;
+use proc_macro::{EscapeError, Literal, ConversionErrorKind};
 
 pub fn test() {
     test_display_literal();
@@ -111,4 +111,17 @@ fn test_literal_value() {
     {
         assert_eq!(Literal::f32_unsuffixed(15.).f16_value().map(|f| f.to_string()), Ok("15".to_string()));
     }
+
+    assert_eq!(
+        Literal::character('A').byte_character_value(),
+        Err(ConversionErrorKind::InvalidLiteralKind),
+    );
+    assert_eq!(
+        Literal::character('é').byte_character_value(),
+        Err(ConversionErrorKind::InvalidLiteralKind),
+    );
+    assert_eq!(
+        Literal::byte_character(b'B').byte_character_value(),
+        Ok(b'B')
+    );
 }

--- a/tests/ui/proc-macro/auxiliary/api/proc_macro_api_tests.rs
+++ b/tests/ui/proc-macro/auxiliary/api/proc_macro_api_tests.rs
@@ -4,6 +4,7 @@
 #![feature(proc_macro_value)]
 #![feature(f16)]
 #![feature(cfg_target_has_reliable_f16_f128)]
+#![feature(rustc_private)]
 #![deny(dead_code)] // catch if a test function is never called
 
 extern crate proc_macro;

